### ISSUE TITLE
Improve accuracy of unit-testing

### ISSUE THREAD

**Issue by frapac** [andrewrosemberg/HydroPowerModels.jl#13]
At first sight, the code coverage seems very low (25%), but you did a lot of appreciable efforts to test correctly HydroPowerModels.jl. Investigating it further with [codecov](https://codecov.io/gh/andrewrosemberg/HydroPowerModels.jl/tree/master/src), it occurs that most files are perfectly tested, except: 
- `IO.jl` (75%)
- `utilities.jl` ( 43%)
- and `visualize_data.jl` (0%)

The code coverage is low because of this last file. Do you think it could be relevant to move it a separate folder, dedicated to the plotting utilities (as `src/visualization` in the package [SDDP.jl](https://github.com/odow/SDDP.jl/tree/master/src/visualization))? 

Another question is: why are only the deterministic examples tested in `runtests.jl`? I would appreciate having a testing procedure that tests stochastic and non-linear examples as well. My opinion is that running a non-linear version of SDDP is one of the key ideas in HydroPowerModels.jl, and you might want to put more emphasis on it. 

X-ref https://github.com/JuliaCon/proceedings-review/issues/35

**Comment by andrewrosemberg** [comment]
> The code coverage is low because of this last file. Do you think it could be relevant to move it a separate folder, dedicated to the plotting utilities (as src/visualization in the package SDDP.jl)?

Yes this is a good suggestion! Added in 250becbcb589eb0b2bf3ac71b6797bb714ca4ec1 .

> I would appreciate having a testing procedure that tests stochastic and non-linear examples as well. My opinion is that running a non-linear version of SDDP is one of the key ideas in HydroPowerModels.jl, and you might want to put more emphasis on it.

Case 3 is a small but stochastic case. Just added a SOC stochastic case in 250becbcb589eb0b2bf3ac71b6797bb714ca4ec1.
